tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 422, results.text
+
+
+@skip_for_ha
+def test_32_try_to_create_an_encrypted_root_with_key_on_passphrase_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'key': dataset_token_hex,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 422, results.text
+
+
+@skip_for_ha
+def test_33_delete_the_passphrase_encrypted_pool_with_is_datasets():
+    payload = {
+        'cascade': True,
+        'restart_services': True,
+        'destroy': True
+    }
+    results = POST(f'/pool/id/{pool_id}/export/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_34_creating_a_key_encrypted_pool():
+    global pool_id
+    payload = {
+        'name': pool_name,
+        'encryption': True,
+        'encryption_options': {
+            'algorithm': 'AES-128-CCM',
+            'key': pool_token_hex,
+        },
+        'topology': {
+            'data': [
+                {'type': 'STRIPE', 'disks': disk_pool}
+            ],
+        }
+    }
+    results = POST('/pool/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    pool_id = job_status['results']['result']['id']
+
+
+@skip_for_ha
+def test_35_creating_a_key_encrypted_root_on_key_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'key': dataset_token_hex,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_36_change_a_key_encrypted_root_to_passphrase_on_key_encrypted_pool():
+    payload = {
+        'id': dataset,
+        'change_key_options': {
+            'passphrase': 'my_passphrase',
+        }
+    }
+    results = POST(f'/pool/dataset/change_key/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_37_lock_passphrase_encrypted_dataset():
+    payload = {
+        'id': dataset,
+        'lock_options': {
+            'force_umount': True
+        }
+    }
+    results = POST('/pool/dataset/lock', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_38_verify_passphrase_encrypted_root_is_locked():
```

Comment:
We don't need to use `encryption_summary` to verify if it's locked, simply retrieving the data for the dataset in question will retrieve this information via `locked` attribute

Reply:
However, as above we could probably use this call to ensure that `encryption_summary` does say that the dataset can be unlocked with the right passphrase